test/e2e/fpga/fpga.go

```diff
@@ -0,0 +1,143 @@
+// Copyright 2020 Intel Corporation. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2efpga
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/intel/intel-device-plugins-for-kubernetes/test/e2e/utils"
+	"github.com/onsi/ginkgo"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2epod "k8s.io/kubernetes/test/e2e/framework/pod"
+)
+
+const (
+	serviceAccountYaml   = "deployments/fpga_plugin/fpga_plugin_service_account.yaml"
+	pluginYaml           = "deployments/fpga_plugin/fpga_plugin.yaml"
+	webhookDeployScript  = "scripts/webhook-deploy.sh"
+	nlb0NodeResource     = "fpga.intel.com/af-f7df405cbd7acf7222f144b0b93acd18"
+	nlb0PodResource      = "fpga.intel.com/arria10.dcp1.2-nlb0"
+	nlb3PodResource      = "fpga.intel.com/arria10.dcp1.2-nlb3"
+	arria10dcp12Resource = "fpga.intel.com/region-69528db6eb31577a8c3668f9faa081f6"
+)
+
+func init() {
+	ginkgo.Describe("FPGA Plugin E2E tests", describe)
+}
+
+func describe() {
+	// Cleanup: delete plugin daemonset
+	// It's not deleted by ginkgo as it's in the kube-system namespace
```

Comment:
I don't recall what the conclusion was about `kube-system` for FPGA. We dropped this setting from GPU and made it so that we have a kustomization to add `kube-system` if that is desired.

Reply:
Thank you for the review. intel-fpga-plugin-controller ServiceAccount requires namespace. We can try to avoid using it like gnu plugin does or at least set the namespace for the ServiceAccount the same way as webhook does.

Reply:
@rojkov can you remind us what was the reason of using ServiceAccount for FPGA plugin? It seems that it's the only plugin that uses it? Is it somehow required by the webhook?

Reply:
IIRC you need ServiceAccount when accessing the api-server with clientset like here
https://github.com/intel/intel-device-plugins-for-kubernetes/blob/33f09160e027081f1f18d90a6393748fb7f1da05/cmd/fpga_plugin/mode.go#L34

Reply:
@rojkov Can use default service account for that? Why do we need to create special intel-fpga-plugin-controller ServiceAccount?

Reply:
Honestly I don't remember though I do remember I omitted the account settings first because I didn't know of its existence (and it didn't work).

Probably the default account doesn't (didn't) have the needed permission ("get node").

Reply:
It seems it has. At least fpga plugin seems to start without a warning in default namespace with these changes:
```
--- a/deployments/fpga_plugin/fpga_plugin.yaml
+++ b/deployments/fpga_plugin/fpga_plugin.yaml
@@ -2,7 +2,7 @@ apiVersion: apps/v1
 kind: DaemonSet
 metadata:
   name: intel-fpga-plugin
-  namespace: kube-system
+  namespace: default
   labels:
     app: intel-fpga-plugin
 spec:
@@ -14,7 +14,7 @@ spec:
       labels:
         app: intel-fpga-plugin
     spec:
-      serviceAccountName: intel-fpga-plugin-controller
+      serviceAccountName: default
       initContainers:
       - name: intel-fpga-initcontainer
         image: intel/intel-fpga-initcontainer:devel
diff --git a/deployments/fpga_plugin/fpga_plugin_service_account.yaml b/deployments/fpga_plugin/fpga_plugin_service_account.yaml
index 261d8be..fecc5c6 100644
--- a/deployments/fpga_plugin/fpga_plugin_service_account.yaml
+++ b/deployments/fpga_plugin/fpga_plugin_service_account.yaml
@@ -1,9 +1,4 @@
 apiVersion: v1
-kind: ServiceAccount
-metadata:
-  name: intel-fpga-plugin-controller
-  namespace: kube-system
----
 kind: ClusterRole
 apiVersion: rbac.authorization.k8s.io/v1
 metadata:
@@ -17,11 +12,11 @@ kind: ClusterRoleBinding
 apiVersion: rbac.authorization.k8s.io/v1
 metadata:
   name: get-nodes
-  namespace: kube-system
+  namespace: default
 subjects:
 - kind: ServiceAccount
-  name: intel-fpga-plugin-controller
-  namespace: kube-system
+  name: default
+  namespace: default
 roleRef:
   kind: ClusterRole
   name: node-getter
```

Reply:
However, having a separate ServiceAccount is not that bad. We just need to use the same approach for plugin and webhook - either use default service account or create our own.

Reply:
Yeap, I'm ok either way.